src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,50 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        MachineProducer<T> producer = null;
+        if (type == Bulldozer.class) {
+            producer = (MachineProducer<T>) new BulldozerProducer();
+        }
+        if (type == Truck.class) {
+            producer = (MachineProducer<T>) new TruckProducer();
+        }
+        if (type == Excavator.class) {
+            producer = (MachineProducer<T>) new ExcavatorProducer();
+        }
+        if (producer != null) {
+            return (List<T>) producer.get();
+        } else {
+            return List.of();
+        }
+    }
+
+    @Override
+    public void fill(List<? super T> machines, T value) {
+        int size = machines.size();
+        for (int i = 0; i < size; i++) {
+            machines.set(i, value);
+        }
+
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> machines) {
```

Comment:
There is an unnecessary null check in the 'startWorking' method. The method should assume that it is called with a non-null list of machines. If a null check is needed, it should be done before calling this method.